src/sentry/api/endpoints/integrations/organization_integrations/index.py

```diff
@@ -1,60 +1,89 @@
+from __future__ import annotations
+
+from typing import Any, Mapping, Sequence
+
 from rest_framework.request import Request
 from rest_framework.response import Response
 
 from sentry.api.bases.organization import OrganizationEndpoint, OrganizationIntegrationsPermission
 from sentry.api.paginator import OffsetPaginator
 from sentry.api.serializers import serialize
-from sentry.models import ObjectStatus, OrganizationIntegration
+from sentry.models import ObjectStatus, Organization, OrganizationIntegration
+
+
+def prepare_feature_filters(features_raw: Sequence[str]) -> set[str]:
+    """Normalize feature names from query params."""
+    return {feature.lower().strip() for feature in features_raw}
+
+
+def prepare_features(organization_integration: OrganizationIntegration) -> set[str]:
+    """Normalize feature names Integration provider feature lists."""
+    return {
+        feature.name.lower().strip()
+        for feature in organization_integration.integration.get_provider().features
+    }
+
+
+def filter_by_features(
+    organization_integrations: Sequence[OrganizationIntegration],
+    feature_filters: Sequence[str],
+) -> Sequence[OrganizationIntegration]:
+    """Filter the list of organization integrations by feature."""
+    return [
+        organization_integration
+        for organization_integration in organization_integrations
+        if prepare_feature_filters(feature_filters).intersection(
+            prepare_features(organization_integration)
+        )
+    ]
 
 
 class OrganizationIntegrationsEndpoint(OrganizationEndpoint):
     permission_classes = (OrganizationIntegrationsPermission,)
 
-    def get(self, request: Request, organization) -> Response:
+    def get(self, request: Request, organization: Organization) -> Response:
+        """
+        List the available Integrations for an Organization
+        ```````````````````````````````````````````````````
 
-        # filter by integration provider features
-        features = [feature.lower() for feature in request.GET.getlist("features", [])]
+        :pparam string organization_slug: The slug of the organization.
 
-        # show disabled org integrations but not ones being deleted
-        integrations = OrganizationIntegration.objects.filter(
+        :qparam string provider_key: Filter by specific integration provider. (e.g. "slack")
+        :qparam string[] features: Filter by integration features names.
+        :qparam bool includeConfig: Should integrations configurations be fetched from third-party
+            APIs? This can add several thousand ms to the request round trip.
```

Comment:
nit: I'd prefer "several seconds", since 'thousand miliseconds' is sorta redundant. Though that's just my opinion.